### PR TITLE
Added zoom service

### PR DESCRIPTION
**Pull request by BogdanBida**

### CODE REVIEW

**Comment by BogdanBida** [comment]
I tried not to make many changes in `canv.component` in this branch to avoid possible conflicts

File: src/app/core/services/map-zoom.service.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { environment } from '../../../environments/environment';
```
Comment (Tornik73) [inline src/app/core/services/map-zoom.service.ts:4]: Please use `src` path

File: src/app/core/services/map-zoom.service.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+const MAX_ZOOM = 1;
```
Comment (Tornik73) [inline src/app/core/services/map-zoom.service.ts:6]: Please move to constants

File: src/app/core/services/map-zoom.service.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  public readonly isMaxZoom$ = new BehaviorSubject<boolean>(
+    DEFAULT_ZOOM_FACTOR === MAX_ZOOM
+  );
+
+  public readonly isMinZoom$ = new BehaviorSubject<boolean>(
+    DEFAULT_ZOOM_FACTOR === MIN_ZOOM
+  );
+
+  public readonly isOneHungredPercentZoom$ = new BehaviorSubject<boolean>(
+    DEFAULT_ZOOM_FACTOR * ONE_HUNGRED === ONE_HUNGRED
```
Comment (Tornik73) [inline src/app/core/services/map-zoom.service.ts:29]: Also check we can move this calculations to constants or utils, and please check can we rename `ONE_HUNGRED` constant

File: src/app/core/services/map-zoom.service.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  );
+
+  public readonly scale$ = this.zoomFactor$.pipe(
+    map((value) => Math.round(value * ONE_HUNGRED))
```
Comment (Tornik73) [inline src/app/core/services/map-zoom.service.ts:33]: Please move to utils

File: src/app/core/services/map-zoom.service.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+@Injectable({
+  providedIn: 'root',
+})
+export class MapZoomService {
+  public readonly zoomFactor$ = new BehaviorSubject<number>(
+    DEFAULT_ZOOM_FACTOR
+  );
+
+  public readonly isMaxZoom$ = new BehaviorSubject<boolean>(
+    DEFAULT_ZOOM_FACTOR === MAX_ZOOM
```
Comment (Tornik73) [inline src/app/core/services/map-zoom.service.ts:21]: Please check, we need move it to utils, and remove code duplication
Also for `DEFAULT_ZOOM_FACTOR === MIN_ZOOM`

File: src/app/core/services/map-zoom.service.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  public readonly scale$ = this.zoomFactor$.pipe(
+    map((value) => Math.round(value * ONE_HUNGRED))
+  );
+
+  public readonly scaleTransform$ = this.zoomFactor$.pipe(
+    map((value) => `scale(${value})`)
+  );
+
+  public zoomIn(step: number = DEFAULT_ZOOM_STEP): void {
+    const value = this._zoomFactor;
```
Comment (Tornik73) [inline src/app/core/services/map-zoom.service.ts:41]: Please check, do we really need this constant `const value = this._zoomFactor;` ?

File: src/app/core/services/map-zoom.service.ts [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  public resetZoomFactor(): void {
+    this._zoomFactor = DEFAULT_ZOOM_FACTOR;
+  }
+
+  private get _zoomFactor(): number {
+    return this.zoomFactor$.getValue();
+  }
+
+  private set _zoomFactor(rawValue: number) {
+    const value = Math.round(rawValue * ONE_HUNGRED) / ONE_HUNGRED;
```
Comment (Tornik73) [inline src/app/core/services/map-zoom.service.ts:63]: Please check, do we really need to do this opertaion?
And rawValue can be float?

File: src/app/core/services/map-zoom.service.ts [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+    if (value > MAX_ZOOM || value < MIN_ZOOM) {
```
Comment (Tornik73) [inline src/app/core/services/map-zoom.service.ts:65]: Please move it to setter constant `value > MAX_ZOOM || value < MIN_ZOOM`

File: src/app/feature/map/canva/canva.component.ts [hunk 9]
```diff
@@ -105,8 +114,9 @@ export class CanvaComponent implements OnInit, AfterViewInit {
   ): void {
     if (left && top) {
       // adapt values ​​for zoom
-      top = top * this.zoomFactor + (HEIGHT - HEIGHT * this.zoomFactor) / 2;
-      left = left * this.zoomFactor + (WIDTH - WIDTH * this.zoomFactor) / 2;
+      const zoomFactor = this._mapZoomService.zoomFactor$.getValue();
```
Comment (Tornik73) [inline src/app/feature/map/canva/canva.component.ts:117]: Please move this logic to service

File: src/app/feature/map/ui-controllers/zoom-controls/zoom-controls.component.html [hunk 10]
```diff
@@ -1,13 +1,13 @@
...
       <use xlink:href="assets/icons/sprite.svg#zoom-in"></use>
     </svg>
   </button>
-  <div class="zoom-value" [class.hidden]="zoomFactor === 1">
-    {{ zoomFactor * 100 }}
+  <div class="zoom-value" [class.hidden]="isOneHungredPercentZoom$ | async">
+    {{ scale$ | async }}
   </div>
-  <button (click)="zoomOut()" [class.disabled]="isMin">
+  <button (click)="zoomOut()" [class.disabled]="isMin$ | async">
```
Comment (Tornik73) [inline src/app/feature/map/ui-controllers/zoom-controls/zoom-controls.component.html:10]: Please move this button to component

**Review by Tornik73 — APPROVED** [review]
Feedbacked
Please apply comments and merge

**Pull request merged**